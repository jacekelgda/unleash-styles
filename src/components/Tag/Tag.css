.Tag {
	display: inline-block;
	margin-bottom: 5px;
}
@media screen and (max-width: 767px) {
	.Tag {
		margin-left: 6px;
	}
}
@media screen and (min-width: 768px) {
	.Tag:not(:last-child) {
		margin-right: 6px;
	}
}

.Tag--hidden {
	display: none;
}

.TagLink {
	display: inline-block;
	font-size: 13px;
	text-decoration: none;
	border: none;
	color: #f33c57;
	background-color: transparent;
	border: 1px solid #e9e9e9;
	border-radius: 2px;
	padding: .25em .75em;
	margin: 0;
	cursor: pointer;
}
