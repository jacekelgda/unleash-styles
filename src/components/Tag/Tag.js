import React from 'react'
import PropTypes from 'prop-types'
import styles from './Tag.css'

/**
 * Tag element on tags component
 */
function Tag (props) {
  return (
    <li className={styles.Tag}>
      <a href={props.href} className={styles.TagLink}>{props.text}</a>
    </li>
  )
}

Tag.propTypes = {
  /** link of the tag */
  href: PropTypes.string.isRequired,
  /** text of the tag */
  text: PropTypes.string.isRequired
}

export default Tag
