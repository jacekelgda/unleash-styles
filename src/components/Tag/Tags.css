.Tags {
	margin: 0;
	padding: 0;
	list-style: none;
	display: inline-flex;
	position: relative;
}
@media screen and (max-width: 767px) {
	.Tags {
		margin-top: 30px;
	}
}

/* hide/show tags via /js/skill-tags.js */
.Tags,
.Tags--unfolded {
	flex-wrap: wrap;
}
@media screen and (max-width: 767px) {
	.Tags,
	.Tags--unfolded {
		justify-content: flex-end;
		padding-right: 20px;
	}
}
.Tags--unfolded .Tag {
	display: inline-block;
}

.Tags::after {
	content: attr(data-content);
	position: absolute;
	right: -15px;
	top: 3px;
	color: #f33c57;
	font: 700 13px 'Open Sans', sans-serif;
	cursor: pointer;
}
