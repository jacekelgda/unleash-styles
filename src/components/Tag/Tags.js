import React from 'react'
import styles from './Tags.css'

/**
 * Tags
 */
function Tags (props) {
  return (
    <ul className={styles.Tags}>
      {props.children}
    </ul>
  )
}

export default Tags
