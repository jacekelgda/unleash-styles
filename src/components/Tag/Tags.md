Tag component:

    <Tag href="http://google.com" text="Foo" />

Tags component:

    <Tags>
      <Tag href="http://google.com" text="one" />
      <Tag href="http://google.com" text="two" />
      <Tag href="http://google.com" text="three" />
    </Tags>
