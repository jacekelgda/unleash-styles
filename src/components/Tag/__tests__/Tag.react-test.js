/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import Tags from '../Tags'
import Tag from '../Tag'

test('Tags renders correctly', () => {
  const wrapper = shallow(
    <Tags>
      <Tag href='http://google.com' text='one' />
      <Tag href='http://google.com' text='two' />
      <Tag href='http://google.com' text='three' />
    </Tags>
  )
  expect(wrapper).toMatchSnapshot()
})
