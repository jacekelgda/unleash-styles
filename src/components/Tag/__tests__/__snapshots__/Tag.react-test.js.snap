// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="Tags"
>
    <Tag
        href="http://google.com"
        text="one"
    />
    <Tag
        href="http://google.com"
        text="two"
    />
    <Tag
        href="http://google.com"
        text="three"
    />
</ul>,
  "nodes": Array [
    <ul
      className="Tags"
>
      <Tag
            href="http://google.com"
            text="one"
      />
      <Tag
            href="http://google.com"
            text="two"
      />
      <Tag
            href="http://google.com"
            text="three"
      />
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Tags>
        <Tag
                href="http://google.com"
                text="one"
        />
        <Tag
                href="http://google.com"
                text="two"
        />
        <Tag
                href="http://google.com"
                text="three"
        />
</Tags>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <Tag
              href="http://google.com"
              text="one"
/>,
            <Tag
              href="http://google.com"
              text="two"
/>,
            <Tag
              href="http://google.com"
              text="three"
/>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="Tags"
>
          <Tag
                    href="http://google.com"
                    text="one"
          />
          <Tag
                    href="http://google.com"
                    text="two"
          />
          <Tag
                    href="http://google.com"
                    text="three"
          />
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="Tags"
>
          <Tag
                    href="http://google.com"
                    text="one"
          />
          <Tag
                    href="http://google.com"
                    text="two"
          />
          <Tag
                    href="http://google.com"
                    text="three"
          />
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tags>
    <Tag
        href="http://google.com"
        text="one"
    />
    <Tag
        href="http://google.com"
        text="two"
    />
    <Tag
        href="http://google.com"
        text="three"
    />
</Tags>,
}
`;
